web/client-api/src/main/java/io/deephaven/web/client/api/filter/FilterValue.java

```diff
@@ -45,20 +50,54 @@ public static FilterValue ofString(@TsTypeRef(Any.class) Object input) {
         return new FilterValue(lit);
     }
 
+    @TsUnion
+    @JsType(name = "?", namespace = JsPackage.GLOBAL, isNative = true)
+    public interface OfNumberUnionParam {
+        @JsOverlay
+        static OfNumberUnionParam of(@DoNotAutobox Object value) {
+            return Js.cast(value);
+        }
+
+        @JsOverlay
+        default boolean isLongWrapper() {
+            return this instanceof LongWrapper;
+        }
+
+        @JsOverlay
+        default boolean isNumber() {
+            return (Object) this instanceof Double;
+        }
+
+        @TsUnionMember
+        @JsOverlay
+        default LongWrapper asLongWrapper() {
+            return Js.cast(this);
+        }
+
+        @TsUnionMember
+        @JsOverlay
+        default double asNumber() {
+            return Js.asDouble(this);
+        }
+    }
+
     @JsMethod(namespace = "dh.FilterValue")
-    public static FilterValue ofNumber(@TsTypeRef(Any.class) Object input) {
+    public static FilterValue ofNumber(OfNumberUnionParam input) {
         Objects.requireNonNull(input);
-        if (input instanceof DateWrapper) {
-            Literal lit = new Literal();
-            lit.setNanoTimeValue(((DateWrapper) input).valueOf());
-            return new FilterValue(lit);
-        } else if (input instanceof LongWrapper) {
-            Literal lit = new Literal();
-            lit.setLongValue(((LongWrapper) input).valueOf());
-            return new FilterValue(lit);
-        } else if (Js.typeof(input).equals("number")) {
+        if (input.isLongWrapper()) {
+            LongWrapper value = input.asLongWrapper();
+            if (value instanceof DateWrapper) {
```

Comment:
Yes? JS doesn't have a cast, but this does follow what instanceof does in plain JS as well - DateWrapper extends from LongWrapper, so if we first pass the `instanceof LongWrapper` check, we might potentially also be an instance of `DateWrapper`. Or am I missing something?

Reply:
Just wasn't clear to me how the `.asLongWrapper()` worked

Reply:
Ah ok - it really just is a helper that does a plain `instanceof` check - the only trick is that it is an "overlay method", so it isn't actually part of any class, it is actually a static method that has an extra argument for "this", so that you can call it on anything that you cast to this type. That way we treat it like a wrapper on whatever was passed from JS, and can check the contents and "unwrap" (though the unwrap is really just a cast), but instead of explicit instanceofs and casts, we have a concrete type that  we can statically examine, and generate ts from.